lib/charms/tempo_k8s/v2/tracing.py

```diff
@@ -590,10 +595,11 @@ def publish_receivers(self, receivers: Sequence[RawReceiver]):
             try:
                 TracingProviderAppData(
                     host=self._host,
-                    external_url=self._external_url,
+                    external_url=f"http://{self._external_url}" if self._external_url else None,
```

Comment:
ok, so this is definitely wrong.
What's happening is: traefik-route is giving us a 'scheme' but we're completely ignoring it.

![image](https://github.com/canonical/tempo-k8s-operator/assets/6230162/6331d829-324a-46c1-af1c-54d46f5dc963)

what we should be doing is:
![image](https://github.com/canonical/tempo-k8s-operator/assets/6230162/c762fe11-4323-4036-a449-443ad126dafe)

and then:

![image](https://github.com/canonical/tempo-k8s-operator/assets/6230162/fbba6ec6-be88-4928-9b18-c7dfaf10632f)

Reply:
see #98 for my current progress on this